lib/crate_ruby/client.rb

```diff
@@ -230,7 +234,12 @@ def request(req, http_options = {})
     def headers
       header = { 'Content-Type' => 'application/json' }
       header['Default-Schema'] = schema if schema
+      header['Authorization'] =  "Basic #{encrypted_credentials}" if username && password
```

Comment:
basic auth should also work without password
see https://github.com/crate/crate-python/blob/master/src/crate/client/http.py#L114-L123

Reply:
also keep backwards compatibility with X-User header